src/main/java/com/amazon/opendistroforelasticsearch/knn/plugin/rest/RestKNNWarmupHandler.java

```diff
@@ -0,0 +1,88 @@
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.knn.plugin.rest;
+
+import com.amazon.opendistroforelasticsearch.knn.common.exception.KNNInvalidIndexException;
+import com.amazon.opendistroforelasticsearch.knn.plugin.KNNPlugin;
+import com.amazon.opendistroforelasticsearch.knn.plugin.transport.KNNWarmupAction;
+import com.amazon.opendistroforelasticsearch.knn.plugin.transport.KNNWarmupRequest;
+import com.google.common.collect.ImmutableList;
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.RestController;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.action.RestToXContentListener;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static com.amazon.opendistroforelasticsearch.knn.index.KNNSettings.KNN_INDEX;
+import static org.elasticsearch.action.support.IndicesOptions.strictExpandOpen;
+
+/**
+ * RestHandler for k-NN index warmup API. API provides the ability for a user to load specific indices' k-NN graphs
+ * into memory.
+ */
+public class RestKNNWarmupHandler extends BaseRestHandler {
+    public static String NAME = "knn_warmup_action";
+
+    private IndexNameExpressionResolver indexNameExpressionResolver;
+    private ClusterService clusterService;
+
+    public RestKNNWarmupHandler(Settings settings, RestController controller, ClusterService clusterService,
+                                IndexNameExpressionResolver indexNameExpressionResolver) {
+        this.clusterService = clusterService;
+        this.indexNameExpressionResolver = indexNameExpressionResolver;
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public List<Route> routes() {
+        return ImmutableList.of(
+                new Route(RestRequest.Method.GET, KNNPlugin.KNN_BASE_URI + "/warmup/{index}")
+        );
+    }
+
+    @Override
+    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) {
+        KNNWarmupRequest knnWarmupRequest = createKNNWarmupRequest(request);
+        return channel -> client.execute(KNNWarmupAction.INSTANCE, knnWarmupRequest, new RestToXContentListener<>(channel));
+    }
+
+    private KNNWarmupRequest createKNNWarmupRequest(RestRequest request) {
+        String[] indexNames = Strings.splitStringByCommaToArray(request.param("index"));
```

Comment:
if indexNames empty, return from here with proper log message?

Reply:
Will add this check.

Reply:
Actually, because I only register [this route](https://github.com/opendistro-for-elasticsearch/k-NN/pull/162/files#diff-9b2b661b10b3b79228d72391a36fd8edR64), it is impossible to pass in no indices. Instead, it fails with the following:

```
curl ${HOST_NAME}/_opendistro/_knn/warmup?pretty
{
  "error" : {
    "root_cause" : [
      {
        "type" : "illegal_argument_exception",
        "reason" : "request [/_opendistro/_knn/warmup] contains unrecognized parameter: [nodeId]"
      }
    ],
    "type" : "illegal_argument_exception",
    "reason" : "request [/_opendistro/_knn/warmup] contains unrecognized parameter: [nodeId]"
  },
  "status" : 400
}
```

It actually does not reach this rest handler. Is this okay?

Reply:
makes sense. Thanks